ssm_jax/nlgssm/extended_inference.py

```diff
@@ -0,0 +1,165 @@
+import chex
+import jax.numpy as jnp
+import jax.random as jr
+from jax import lax
+from jax import jacfwd
+from distrax import MultivariateNormalFullCovariance as MVN
+from typing import Callable
+
+@chex.dataclass
+class ESSMParams:
```

Comment:
Maybe call this `NLGSSMParams`.
Also this should probably live in the nlgssm/models.py file, since it will be shared between EKF, UKF, etc.

Reply:
Ah, since it's so lightweight I was thinking of having a separate Params dataclass for each nlgssm algorithm b/c each algorithm requires a slightly different combination of parameters (e.g. UKF requires alpha, beta, kappa, etc.), but you're right in that I should maybe factor out the common parameters into nlgssm/models.py and extend the dataclass for each algorithm.